src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -170,9 +171,40 @@ public async Task<SecretResponseModel> UpdateSecretAsync([FromRoute] Guid id, [F
     [HttpPost("secrets/delete")]
     public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids)
     {
-        var userId = _userService.GetProperUserId(User).Value;
-        var results = await _deleteSecretCommand.DeleteSecrets(ids, userId);
-        var responses = results.Select(r => new BulkDeleteResponseModel(r.Item1.Id, r.Item2));
+        var secrets = (await _secretRepository.GetManyByIds(ids)).ToList();
+        if (!secrets.Any() || secrets.Count != ids.Count)
+        {
+            throw new NotFoundException();
+        }
+
+        // Ensure all secrets belong to the same organization.
+        var organizationId = secrets.First().OrganizationId;
+        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
+            !_currentContext.AccessSecretsManager(organizationId))
+        {
+            throw new NotFoundException();
```

Comment:
Should probably be a bad request exception.

Reply:
We haven't passed through all the authorization checks at this point, and we had to fetch data to check they're all in the same organization. 

I thought bad request would be inappropriate since we aren't checking the request data, but instead fetching server data for the request data sent to check.